packages/react-aria-components/docs/ComboBox.mdx

```diff
@@ -1033,72 +1051,96 @@ function MyListBox(props) {
+      {/*- begin highlight -*/}
+      <ComboBoxContext.Provider value={{isDisabled}}>
+      {/*- end highlight -*/}
+        {children}
+      </ComboBoxContext.Provider>
+    </fieldset>
+  );
+}
+
+<FieldGroup title="Filters" isDisabled>
+  <MyComboBox label="Status" defaultSelectedKey="published">
+    <Item id="draft">Draft</Item>
+    <Item id="published">Published</Item>
+    <Item id="deleted">Deleted</Item>
+  </MyComboBox>
+  <MyComboBox label="Author" defaultSelectedKey="emma">
+    <Item id="john">John</Item>
+    <Item id="emma">Emma</Item>
+    <Item id="tim">Tim</Item>
+  </MyComboBox>
+</FieldGroup>
+```
 
-  // Get state sent from ComboBox via context, and call useListBox.
-  let {state} = React.useContext(ListBoxContext);
-  let {listBoxProps} = useListBox(props, state, ref);
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
 
-  // Render stuff
-  // ...
+```css
+fieldset {
+  padding: 1.5em;
+  width: fit-content;
 }
 ```
 
-This also works the other way. If you need to customize `ComboBox` itself, but want to reuse the components it contains, you can do so by providing
-the necessary contexts. The `Provider` component is an easier way to send multiple contexts at once.
+</details>
+
+### Custom children
+
+ComboBox passes props to its child components, such as the label and input, via their associated contexts. These contexts are exported so you can also consume them in your own custom components. This enables you to reuse existing components from your app or component library together with React Aria Components.
+
+<ContextTable components={['Label', 'Button', 'Input', 'Popover', 'ListBox', 'Text']} docs={docs} />
+
+This example consumes from `LabelContext` in an existing styled label component to make it compatible with React Aria Components. The <TypeLink links={docs.links} type={docs.exports.useContextProps} /> hook merges the local props and ref with the ones provided via context by ComboBox.
 
 ```tsx
-import {useComboBox, LabelContext, ButtonContext, InputContext, PopoverContext, ListBoxContext} from 'react-aria';
-import {Provider} from 'react-aria-components';
-
-function MyComboBox(props) {
-  // ...
-  let buttonRef = useRef(null);
-  let inputRef = useRef(null);
-  let listBoxRef = useRef(null);
-  let popoverRef = useRef(null);
-  let {
-    buttonProps,
-    inputProps,
-    listBoxProps,
-    labelProps
-  } = useComboBox({ /* ... */ });
+import type {LabelProps} from 'react-aria-components';
+import {LabelContext, useContextProps} from 'react-aria-components';
+
+const MyCustomLabel = React.forwardRef((props: LabelProps, ref: React.ForwardedRef<HTMLLabelElement>) => {
+  // Merge the local props and ref with the ones provided via context.
+  ///- begin highlight -///
+  [props, ref] = useContextProps(props, ref, LabelContext);
+  ///- end highlight -///
+
+  // ... your existing Label component
+  return <label {...props} ref={ref} />;
+});
+```
```

Comment:
Bit wary of these example which don't render anything. I worry that they could break due to a simple typo or even be broken and we wouldn't know

Reply:
I guess the output isn't that interesting / not actually styled. The example is more about showing a technique than the actual result. In some of them I don't even include the whole thing (just a `...`).

Reply:
fingers crossed, i answer so many questions where if people read for 2 more seconds they'd realize they weren't dealing with a copy/paste example